### PR TITLE
Fix heisentests in regards to cache

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

File: exla/lib/exla.ex [hunk 1]
```diff
@@ -294,12 +294,13 @@ defmodule EXLA do
 
   ## Examples
 
+      iex> fun = fn a, b -> Nx.add(a, b) end
```
Comment (polvalente) [inline exla/lib/exla.ex:297]: Shouldn't this be the same as &Nx.add/2?

**Review by polvalente — APPROVED** [review]

File: exla/lib/exla.ex (hunk 1 above)
Comment (polvalente) [reply exla/lib/exla.ex:297]: Ok, so from what I gathered the BEAM will treat captures differently than `fn` functions, is that right?

Comment (josevalim) [reply exla/lib/exla.ex:297]: Yeah, an optimization in the future may break it, but they should be distinct.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**